### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by mahammad-mostafa**
- Module `Storage` used for handling JSON files read & write
- Module `Storage` includes two methods (`read_file` / `write_file`)
- Implemented serializing & deserializing for object instances using (`generate_string` / `parse_string`)
- All data is preserved inside three files (`books.json` / `people.json` / `rentals.json`)

### CODE REVIEW

File: book_class.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+require 'JSON'
```
Comment (anagudelogu) [inline book_class.rb]: - [ ] You did a great job working on this milestone! 💯 However, I'm getting the following error:

image

Once I changed it to lowercase `json` it worked fine! 🤓 🔧

File: storage_module.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+require 'JSON'
```
Comment (anagudelogu) [inline storage_module.rb]: - [ ] Here happens the same:

image

Please consider changing it to lowercase! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi Team! @mahammad-mostafa @alexansaa @stino-x, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜  
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by shahadat3669 — APPROVED** [review]
Hi everyone, 

Your project is complete!  There is nothing else to say other than it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ On exit from  app save all the data
✅ One per type of data are storing and named accordingly (e.g. rentals.json)
✅ On startup of the app load all data from files

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by nicupop729** [comment]
@anagudelogu & @shahadat3669  please note that co-authored commits are not acceptable to have while building projects at Micorverse.

Co-authored commits are done by a student who is adding the names and GitHub accounts of the other students while creating the name of the commits.

We cannot accept this activity as done in pair programming, as we cannot get the level of implication for all group members.

If you encounter this type of situation in the future, let's please invalidate the CRR.

Thank you!

**Pull request merged**